src/zmqsubscriber/zmqsubscriber_test.go

```diff
@@ -36,48 +54,126 @@ func waitForZMQBlock(t *testing.T, z *ZMQSubscriber, timeout time.Duration) *typ
 	}
 }
 
-func getTestZMQSubscriber(t *testing.T, env test.TestEnv) *ZMQSubscriber {
-	z, err := NewZMQSubscriber(env.ZmqHost, env.ZmqPort)
-	require.NoError(t, err)
+func setupAndRunZMQSubscriber(t *testing.T, zmqHost string, zmqPort string) (*ZMQSubscriber, error) {
+	z, err := NewZMQSubscriber(zmqHost, zmqPort)
+	if err != nil {
+		return nil, fmt.Errorf("could not create a new ZMQ Subscriber: %s", err)
+	}
 
 	go func() {
 		if err := z.Run(); err != nil {
 			log.Fatalf("ZMQSubscriber exited with error: %s", err)
 		}
 	}()
 
-	return z
+	return z, nil
+}
+
+// waitForBitcoinRPCCookie blocks while waiting for the Bitcoin Core RPC
+// `.cookie` file to be present. Times out after `maxTimeout`.
+func waitForBitcoinRPCCookie() {
+	const maxTimeout = 10 * time.Second
+	const timeoutDelay = 250 * time.Millisecond
+
+	// The environment variable `TEST_INTEGRATION_DIR` is set to a temporay directory created by
+	// the Makefile in the target `test-integration`.
+	integrationTestDir := os.Getenv("TEST_INTEGRATION_DIR")
+	cookiePath := integrationTestDir + "/datadir/regtest/.cookie"
+	cookieExists := false
+	currentTimeout := 0 * time.Second
+
+	fmt.Printf("Waiting for '%s'...\n", cookiePath)
+
+	for !cookieExists {
+
+		if currentTimeout > maxTimeout {
+			fmt.Printf("Timeout (%s) while waiting for '%s'.\n", maxTimeout, cookiePath)
+			os.Exit(1)
+		}
+
+		_, err := os.Stat(cookiePath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				time.Sleep(timeoutDelay)
+				currentTimeout += timeoutDelay
+			} else {
+				fmt.Printf("Could not stat '%s': %s\n.", cookiePath, err)
+				os.Exit(1)
+			}
+		} else {
+			cookieExists = true
+			break
+		}
+	}
 }
 
 func TestZMQSubscriber(t *testing.T) {
-	const waitTimeout = 5 * time.Second
+	if testing.Short() {
+		t.Skip("Skipping " + t.Name() + " since it's not a unit test.")
+	}
+
+	const zmqWaitTimeout = 5 * time.Second
```

Comment:
nit: if always we pass the same timeout anyway (and don't want to make it a global const either) we can make it a variable that is local to the func

Reply:
I think that's what it is now, isn't it? It's just `const` because it should not be modified.

Reply:
I meant local to `waitForForZMQBlock`